dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -690,6 +695,95 @@ export function JsonRpcContextProvider({
       },
     ),
   };
+
+  // -------- NEAR RPC METHODS --------
+
+  const nearRpc = {
+    testSignAndSendTransaction: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        const method = DEFAULT_NEAR_METHODS.NEAR_SIGN_AND_SEND_TRANSACTION
+        const result = await client!.request({
+          topic: session!.topic,
+          chainId,
+          request: {
+            method,
+            params: {
+              transaction: {
+                signerId: address,
+                receiverId: "guest-book.testnet",
+                actions: [{
+                  type: "FunctionCall",
+                  params: {
+                    methodName: "addMessage",
+                    args: { text: "Hello from Wallet Connect!" },
+                    gas: "30000000000000",
+                    deposit: "0",
+                  }
+                }]
+              }
+            },
+          },
+        });
+
+        return {
+          method,
+          address,
+          valid: true,
+          result: JSON.stringify(result),
+        };
+      }
+    ),
+    testSignAndSendTransactions: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        const method = DEFAULT_NEAR_METHODS.NEAR_SIGN_AND_SEND_TRANSACTIONS
+        const result = await client!.request({
+          topic: session!.topic,
+          chainId,
+          request: {
+            method,
+            params: {
+              transactions: [
+                {
+                  signerId: address,
+                  receiverId: "guest-book.testnet",
+                  actions: [{
+                    type: "FunctionCall",
+                    params: {
+                      methodName: "addMessage",
+                      args: { text: "Hello from Wallet Connect! (1/2)" },
+                      gas: "30000000000000",
+                      deposit: "0",
+                    }
+                  }]
+                },
+                {
+                  signerId: address,
+                  receiverId: "guest-book.testnet",
+                  actions: [{
+                    type: "FunctionCall",
+                    params: {
+                      methodName: "addMessage",
+                      args: { text: "Hello from Wallet Connect! (2/2)" },
+                      gas: "30000000000000",
+                      deposit: "0",
+                    }
+                  }]
+                }
+              ],
+            }
+          },
+        });
+
+        return {
+          method,
+          address,
+          valid: true,
+          result: JSON.stringify(result),
```

Comment:
Output is too large for the modal, condensing to only display the transaction keys:

```suggestion
          result: JSON.stringify((result as any).map((r: any) => r.transaction)),
```